src/map/script.c

```diff
@@ -15617,16 +15617,9 @@ static BUILDIN(gethominfo)
 /// getmercinfo <type>[,<char id>];
 static BUILDIN(getmercinfo)
 {
-	int type;
 	struct map_session_data* sd;
-	struct mercenary_data* md;
-
-	type = script_getnum(st,2);
-
 	if (script_hasdata(st,3)) {
-		int char_id = script_getnum(st,3);
-
-		if ((sd = script->charid2sd(st, char_id)) == NULL) {
+		if ((sd = script->charid2sd(st, script_getnum(st,3))) == NULL) {
```

Comment:
missing a space after the `,` also please check the reset of the line you changed for style issues